js/base-tool.js

```diff
@@ -549,6 +549,149 @@ function baseTool () {
+							<td><label for="diagonaltype">Diagonal Type:</label></td>
+							<td><select name="diagonaltype">
+								<option class="squareonly" selected="" value="foure" style="">D&amp;D 5E/4E Compatible</option>
+								<option class="squareonly" value="threefive" style="">Pathfinder/3.5E Compatible</option>
+								<option class="squareonly" value="manhattan" style="">Manhattan</option>
+								<option class="hexonly" value="hex" style="display: none;">Hex Path</option>
+								<option value="pythagorean">Euclidean</option>
+								</select>
+							</td>
+						</tr>
+						<tr>
+							<td><input type="checkbox" name="apply_lightrestrictmove"></td>
+							<td><label for="lightrestrictmove">Dynamic lighting barriers restrict movement:</label></td>
+							<td><input type="checkbox" name="lightrestrictmove"></td>
+						</tr>
+						<tr>
+							<td><input type="checkbox" name="apply_dynamic_lighting_enabled"></td>
+							<td><label for="dynamic_lighting_enabled">Dynamic Lighting Enabled:</label></td>
+							<td><input type="checkbox" name="dynamic_lighting_enabled"></td>
+						</tr>
+						<tr>
+							<td><input type="checkbox" name="apply_daylight_mode_enabled"></td>
+							<td><label for="daylight_mode_enabled">Daylight mode:</label></td>
+							<td><input type="checkbox" name="daylight_mode_enabled"></td>
+						</tr>
+						<tr>
+							<td><input type="checkbox" name="apply_lightupdatedrop"></td>
+							<td><label for="lightupdatedrop">Light Update Drop:</label></td>
+							<td><input type="checkbox" name="lightupdatedrop"></td>
+						</tr>
+					</tbody>
+				</table>
+                <hr>
+				<p style="text-align: right;"><button name="btn-apply" class="btn btn-danger modifier">Apply</button></p>
+				</div>
+				`,
+			dialogFn: () => {
+				$("#d20plus-mass-page-modify").dialog({
+					autoOpen: false,
+					resizable: true,
+					width: 600,
+					height: 800,
+				});
+			},
+			openFn: () => {
+				const $win = $("#d20plus-mass-page-modify");
+				$win.dialog("open");
+
+				const $lst = $win.find(`.list`).empty();
+
+				d20.Campaign.pages.models.forEach(m => {
+					$lst.append(`
+							<label class="import-cb-label import-cb-label--img" data-listid="${m.id}">
+								<input type="checkbox">
+								<img class="import-label__img" src="${m.attributes.thumbnail}">
+								<span class="name col-9">${m.attributes.name}</span>
+								<span style="display: none;" class="page-id">${m.id}</span>
+							</label>
+						`);
+				});
+
+				const pageList = new List("modify-pages-list", {
+					valueNames: ["name", "page-id"],
+				});
+
+				const $cbAll = $win.find(`.select-all`).off("click").click(() => {
+					pageList.items.forEach(it => {
+						$(it.elm).find(`input[type="checkbox"]`).prop("checked", $cbAll.prop("checked"));
+					});
+				});
+
+				function collectCheckedInputs () {
+					const table = document.querySelector("[name=modifications-table]");
```

Comment:
selector too broad -- should be e.g. ``$win.find(` ... `).get()``

Reply:
Thanks for the suggestion, I am learning new things every iteration.
```suggestion
					const table = $win.find("[name=modifications-table]").get(0);
```